app/views/play/common/ChangeLanguageTab.js

```diff
@@ -36,7 +36,12 @@ module.exports = (ChangeLanguageTab = (function () {
       this.utils = utils
       this.codeLanguageObject = utils.getCodeLanguages()
       this.codeFormatObject = utils.getCodeFormats()
-      this.codeFormat = this.options.codeFormat || me.get('aceConfig')?.codeFormat || 'text-code'
+      const defaultCodeFormat = 'text-code'
+      this.codeFormat = this.options.codeFormat || me.get('aceConfig')?.codeFormat || defaultCodeFormat
+      if (this.isJunior && options.level?.get('slug') === 'the-gem') {
+        // let's default to blocks-text for the-gem (first level) in junior so that users can see the blocks
+        this.codeFormat = 'blocks-text'
+      }
```

Comment:
Shouldn't we just set the `defaultCodeFormat` to `blocks-text` if we're on the `the-gem`, but still respect the user's chosen code format if it's passed in via `this.options` or `me.aceConfig`?

Reply:
that is the issue we are trying to work around by checking for the first level. We don't have a different code format for junior and cs courses. Most of our old users will have code format set to text-code because of playing other levels. When they come to junior, we select text-code as default to respect that property thus missing the chance to provide them blocks to play with. 

Ideally, we would want a different code format property for junior. But, we are also planning to introduce blocks in cs1 and other courses so we didn't want to introduce it before we enable that and then plan it better on how code formats should look.